### PR TITLE
Unselected widget color

### PR DESCRIPTION
**Pull request by YourFavouriteOreo**
## Description

Modifying Icon Color behaviour such that the Icon Color for text field ( or any widget that uses the Input Decorator ) would use the unselectedWidgetColor property if defined instead of using the hardcoded existing brightness dependent colors. (black54 and white70)

## Related Issues

#33183 

## Tests

I ran down the possible use cases for this behaviour with multiple text field running different parameters that would affect the behaviour . There was no failure. Otherwise, would appreciate feedback on how to write a proper test for this.
## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by YourFavouriteOreo** [comment]
@HansMuller  Is there something I am missing on why the tool_tests-windows failed?

**Comment by YourFavouriteOreo** [comment]
Fixed It! Ready for Review!

**Comment by YourFavouriteOreo** [comment]
@HansMuller  Any chance I could get a review?

File: packages/flutter/lib/src/material/input_decorator.dart [hunk 1]
```diff
@@ -1870,6 +1870,11 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
     if (!decoration.enabled)
       return themeData.disabledColor;
 
+      // Checks if unselectedWidgetColor property is not one of the defaults and returns based on the result
+    if (themeData.unselectedWidgetColor != Colors.black54??Colors.white70){
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/input_decorator.dart]: This probably isn't what you want, see http://blog.sethladd.com/2015/07/null-aware-operators-in-dart.html

InputDecorator uses `IconTheme` to define the default icon colors for widgets like prefixIcon and suffixIcon. Applications can override the default by creating a widget that includes an Icon that specifies the [icon's color (https://api.flutter.dev/flutter/widgets/Icon/color.html).

Comment (YourFavouriteOreo) [reply packages/flutter/lib/src/material/input_decorator.dart]: I love learning from my mistakes so the more the merrier.

I changed the code as per your spec however I noticed that we get the focused color from seperate sources and not Icon Theme. Should I change it for consistency or should we best leave as is?

File: packages/flutter/lib/src/material/input_decorator.dart [hunk 2]
```diff
@@ -1870,6 +1870,11 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
     if (!decoration.enabled)
       return themeData.disabledColor;
 
+      // Checks if iconTheme.color property is not one of the defaults and returns based on the result
+    if (themeData.iconTheme.color != Colors.black54 && themeData.iconTheme.color != Colors.white70){
```
Comment (chunhtai) [inline packages/flutter/lib/src/material/input_decorator.dart]: I think you mean Colors.black45, but either way I don't think this is the right approach as other developer who sets themeData.iconTheme.color will sometimes hit the switch case and sometimes not, and if user what to force themeData.iconTheme.color to be Colors.white70, they will never be able to do so due the the switch case.

I will suggest we change InputDecoration class to include the iconTheme.

**Review by chunhtai — COMMENTED** [review]
For testing, you can create a Textfield with icontheme set. and use finder to find the IconTheme to make sure it have the right value

**Comment by Saifallak** [comment]
any updates on this?!

**Comment by YourFavouriteOreo** [comment]
@Saifallak hey, my bad. Life got in the way. I'll try implementing this over the course of the next 2 days <3

**Comment by Saifallak** [comment]
@YourFavouriteOreo np, just checking the progress of this. i hopes the best for you, thanks for contributing anyway ^^!

File: packages/flutter/lib/src/material/input_decorator.dart (hunk 2 above)
Comment (YourFavouriteOreo) [reply packages/flutter/lib/src/material/input_decorator.dart]: Sorry, it took me a whole month to see this. Can you explain more on this approach? We could just introduce an unselectedColor as an optional parameter that gets set with a default. Wouldn't that be easier?

Comment (chunhtai) [reply packages/flutter/lib/src/material/input_decorator.dart]: Yes, we should introduce an optional parameter to InputDecoration class that sets the unselected color

**Comment by gspencergoog** [comment]
@YourFavouriteOreo, are you still interested in pursuing this change? You can always close it and come back with an updated change when you're ready.

If do want to proceed, from what I can see, in addition to the review comments, you'll need to address the test failures and add your own tests that fail without your changes, and succeed with them.

**Comment by YourFavouriteOreo** [comment]
@gspencer

> 
> 
> @YourFavouriteOreo, are you still interested in pursuing this change? You can always close it and come back with an updated change when you're ready.
> 
> If do want to proceed, from what I can see, in addition to the review comments, you'll need to address the test failures and add your own tests that fail without your changes, and succeed with them.

I think I will close this and get back to it soon as I can. Since August my life has been swamped sadly and put my own projects on hold . Closed till further notice where my life has more free time .

**Pull request closed without merging**